code/modules/projectiles/updated_projectiles/ammo_datums.dm

```diff
@@ -1369,13 +1369,19 @@
 	drop_resin(get_turf(M))
 	if(istype(M,/mob/living/carbon))
 		var/mob/living/carbon/C = M
-		C.add_slowdown(0.7) //slow em down
+		if( (xeno_hivenumber(C) && xeno_hivenumber(C) == xeno_hivenumber(P.firer) ) )
+			return
+		C.add_slowdown(1) //slow em down
+		C.next_move_slowdown += 8 //really slow down their next move, as if they stepped in sticky doo doo
 
 /datum/ammo/xeno/sticky/on_hit_obj(obj/O,obj/item/projectile/P)
 	drop_resin(get_turf(P))
 
 /datum/ammo/xeno/sticky/on_hit_turf(turf/T,obj/item/projectile/P)
-	drop_resin(get_turf(P))
+	if(istype(T, /turf/closed) )
```

Comment:
This proc will only trigger when the turf is dense and stops projectile movement, or when it's a diagonal shot and Adjacency fails.
So no `istype` check needed, just drop_resin(get_turf(P)) should suffice.

Reply:
Except it didn't, because people were reporting that the resin would stop short

Reply:
I can only assume it considers air a solid block so it stops it one tile before that :P